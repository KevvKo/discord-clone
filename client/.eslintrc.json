{
    "env": {
        "browser": true,
        "es2021": true,
        "node": true
    },
    "extends": [
        "eslint:recommended",
        "plugin:react/recommended"
    ],
    "parserOptions": {
        "ecmaFeatures": {
            "jsx": true
        },
        "ecmaVersion": 12,
        "sourceType": "module"
    },
    "plugins": [
        "react"
    ],
    "globals": {
        "process": "readonly"
    },
    "rules": {
        "indent": [
            "error",
            4
        ],
        "no-unused-vars": "warn",
        "quotes": [
            "error",
            "single"
        ],
        "semi": [
            "error",
            "always"
        ]
    }
}
