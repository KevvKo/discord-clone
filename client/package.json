{
  "name": "client",
  "version": "0.1.0",
  "private": true,
  "homepage": ".",
  "dependencies": {
    "-": "0.0.1",
    "@apollo/client": "^3.3.20",
    "@reduxjs/toolkit": "^1.6.1",
    "@testing-library/jest-dom": "^5.11.4",
    "@testing-library/react": "^11.1.0",
    "@testing-library/user-event": "^12.1.10",
    "bootstrap": "^4.6.0",
    "bootstrap-icons": "^1.5.0",
    "generate-react-component": "^0.1.6",
    "graphql": "^15.5.1",
    "i18next": "^20.3.1",
    "react": "^17.0.2",
    "react-bootstrap": "^1.6.0",
    "react-dom": "^17.0.2",
    "react-i18next": "^11.10.0",
    "react-redux": "^7.2.5",
    "react-router-dom": "^5.2.0",
    "react-scripts": "4.0.3",
    "redux": "^4.1.1",
    "web-vitals": "^1.0.1",
    "workbox-background-sync": "^5.1.3",
    "workbox-broadcast-update": "^5.1.3",
    "workbox-cacheable-response": "^5.1.3",
    "workbox-core": "^5.1.3",
    "workbox-expiration": "^5.1.3",
    "workbox-google-analytics": "^5.1.3",
    "workbox-navigation-preload": "^5.1.3",
    "workbox-precaching": "^5.1.3",
    "workbox-range-requests": "^5.1.3",
    "workbox-routing": "^5.1.3",
    "workbox-strategies": "^5.1.3",
    "workbox-streams": "^5.1.3"
  },
  "scripts": {
    "start": "react-scripts start",
    "dev": "react-scripts",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "proxy": "http://localhost:3001",
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "eslint": "^7.32.0",
    "eslint-plugin-react": "^7.24.0",
    "eslint-plugin-testing-library": "^4.12.4"
  }
}
