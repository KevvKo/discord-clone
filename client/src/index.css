@import './css/colors.css';

body{
  color: var(--body-font-color);
}
