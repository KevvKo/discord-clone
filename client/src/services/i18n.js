import i18next from "i18next";
import { initReactI18next } from "react-i18next";
import commonDe from '../configs/translations/de/common.json'
import commonEn from '../configs/translations/en/common.json'

const resources = {
  en: {
      common: commonEn
  },
  de: {
      common: commonDe
  }
}

i18next
  .use(initReactI18next)
  .init({
    resources,
    lng: "en",
    interpolation: {
      escapeValue: false,
    },
  });

export default i18next;