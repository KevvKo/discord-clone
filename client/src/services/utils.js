// Array 

/**
 * @description returns a sequence as array
 * @param {integer} start 
 * @param {Integer} stop 
 * @param {Integer} step 
 */

export const range = (start, stop, step = 1) => {
        
    return Array.from(
        { length: (stop - start) / step + 1}, 
        (_, i) => start + (i * step)
    );
};