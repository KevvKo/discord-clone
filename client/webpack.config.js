module.exports = {
    module: {
        rules: [
            {
                test: /\.(png|jpg|gif|svg)$/i,
                use: [
                    {
                        loader: 'url-loader',
                        options: {
                            limit: 8192,
                        }
                    },
                ],
        
                type: 'javascript/auto'
            },
        ]
    },
};