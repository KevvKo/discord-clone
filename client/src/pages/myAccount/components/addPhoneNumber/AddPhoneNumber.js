import React from 'react';

// Utilities
import PropTypes from 'prop-types';
// Components
import { Button, InputGroup, Form } from 'react-bootstrap';
import PrefixDropdown from './prefixesDropdown/PrefixesDropdown';
import FormModal from '../../../../components/modals/FormModal/FormModal';
// Hooks 
import { useState, useEffect } from 'react';
import { useTranslation } from 'react-i18next';
import { useMutation, useQuery } from '@apollo/client';
//Utilities
import { ADD_PHONE_NUMBER } from '../../../../graphql/mutations';
import { USER_QUERY } from '../../../../graphql/query';
function AddPhoneNumber(){

    const [ t ] = useTranslation('common');
    const [ show, setShow ] = useState(false);
    const [ prefix, setPrefix ] = useState('');
    const [ phoneNumber, setPhoneNumber ] = useState('');
    const { data } = useQuery(USER_QUERY); 
    const [ addPhoneNumber ] = useMutation(ADD_PHONE_NUMBER, {
        onCompleted: () => {
            window.location.reload();
        },
        onError: (error) => {
            console.log(error.message);
        }
    });

    const handleShow = () => {
        show
            ? setShow(false)
            : setShow(true);
    };
    
    const handleSelect = (e) => {
        setPrefix(e);
    };
    const handleSubmit = (e) => {

        e.preventDefault();
        const form = e.target;
        const { phoneNumber } = form;
        
        if(phoneNumber){
            addPhoneNumber({
                variables: {
                    phoneNumber: prefix + phoneNumber.value,
                }
            });
        }
    };

    useEffect(() => {
        if(data){
            setPhoneNumber(data.user.phoneNumber);
        }
    }, [ data ]);

    const description = phoneNumber
        ?  <span>{phoneNumber}</span>
        :  <span>{t('settings.main.myAccount.noPhoneNumber')}</span>;

    return(
        <>
            <div>
                <h6>{ t('settings.main.myAccount.phoneNumber') }</h6>
                { description }
            </div>
            <Button onClick={ handleShow } className='ml-auto' variant='secondary'>{t('settings.main.myAccount.add')}</Button>
            <FormModal modalTitle={t('settings.main.myAccount.changePhoneNumber')} show={ show } handleOnHide={ handleShow }>
                <p className='text-center'>
                    {t('settings.main.myAccount.editPhoneDescriptionI')}
                </p>
                <p className='mt-2 text-center'>
                    {t('settings.main.myAccount.editPhoneDescriptionII')}
                </p>
                <Form onSubmit={ handleSubmit }>
                    <InputGroup>
                        <PrefixDropdown onSelect= { handleSelect }/>
                        <Form.Control type='tel' name='phoneNumber' ></Form.Control>
                        <Button variant='primary' type='submit'>  {t('settings.main.myAccount.send')} </Button>
                    </InputGroup>
                </Form>
            </FormModal>
        </>
    );
}

AddPhoneNumber.defaultProps = {
    show: false
};

AddPhoneNumber.propTypes = {
    id: PropTypes.string,
    username: PropTypes.string,
    show: PropTypes.bool
};
export default AddPhoneNumber;