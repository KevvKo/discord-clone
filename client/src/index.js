import React from 'react';
import ReactDOM from 'react-dom';
import './index.css';
import App from './app/App';
import * as serviceWorkerRegistration from './serviceWorkerRegistration';
import reportWebVitals from './reportWebVitals';
import 'bootstrap/dist/css/bootstrap.min.css';
import "./services/i18n";
import {
  ApolloProvider,
  ApolloClient,
  createHttpLink,
  InMemoryCache
} from '@apollo/client';

const httpLink = createHttpLink({
  uri: 'http://localhost:4000'
});

const client = new ApolloClient({
  link: httpLink,
  cache: new InMemoryCache()
});

ReactDOM.render(
  <React.StrictMode>
    <ApolloProvider client={client}>
      <App />
    </ApolloProvider>
  </React.StrictMode>,
  document.getElementById('root')
);
// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: https://cra.link/PWA
serviceWorkerRegistration.register();

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals();
