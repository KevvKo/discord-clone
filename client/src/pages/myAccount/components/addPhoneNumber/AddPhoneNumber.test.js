import React from 'react';
import { render, screen, fireEvent, act } from '../../../../scripts/utils';
import renderer from 'react-test-renderer';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { MockedProvider } from '@apollo/client/testing';
import { InMemoryCache } from '@apollo/client';
import userReducer from '../../../../store/slices/userSlice';
import userProfileReducer from '../../../../store/slices/userProfileSlice';
import AddPhoneNumber from './AddPhoneNumber';
import '../../../../scripts/i18n';

describe('AddPhoneNumber component', () => {

    beforeEach(() => {
        render(
            <AddPhoneNumber /> );
    });
    it('renders correctly', () =>{
        const store = configureStore({  reducer: {
            user: userReducer,
            userProfile: userProfileReducer
        }});

        const cache = new InMemoryCache({ });
        const mocks = [];

        const tree = renderer
            .create(
                <MockedProvider cache={cache} mocks={mocks}>
                    <Provider store={store}>
                        <AddPhoneNumber />
                    </Provider>
                </MockedProvider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('render lables correctly', () => {
        expect(screen.getByText(/Phone Number/)).toBeTruthy();
        expect(screen.getByText(/Add/)).toBeTruthy();
        expect(screen.getByText(/no phone number/i)).toBeTruthy();
        expect(screen.queryByText(/Enter Phone Number/i)).toBeNull();
        expect(screen.queryByText(/You will be receiving/i)).toBeNull();
        expect(screen.queryByText(/Send/i)).toBeNull();
    });
    it('should show modal on click', () => {
        fireEvent.click(screen.getByText(/Add/));
        expect(screen.getByText(/Enter Phone Number/i)).toBeTruthy();
        expect(screen.getByText(/You will be receiving/i)).toBeTruthy();
        expect(screen.getByText(/Send/i)).toBeTruthy();
    });
});
