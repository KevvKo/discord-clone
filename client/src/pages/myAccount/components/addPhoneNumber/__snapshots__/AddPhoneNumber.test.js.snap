// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPhoneNumber component renders correctly 1`] = `
Array [
  <div>
    <h6>
      Phone Number
    </h6>
    <span>
      You have no phone number added, yet.
    </span>
  </div>,
  <button
    className="ml-auto btn btn-secondary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Add
  </button>,
]
`;
